### ISSUE TITLE
Tune PyPOTS HDF5 chunking for shuffled per-sample reads

### ISSUE THREAD

**Issue by NarayanSchuetz** [AshleyLab/OpenMHC#16]
## Problem

`src/imputation_training/data_export.py` writes PyPOTS training HDF5 datasets with row chunks shaped `(chunk_size, n_timesteps, 19)`. The default `configs/training/h5_export/default.yaml` sets `chunk_size: 1000`.

With the default `data.n_days=1`, each `X` chunk is approximately:

```text
1000 * 1440 * 19 * 4 bytes = ~109 MB
```

For validation/test, `X_ori` is chunked the same way. For 7-day samples this scales roughly 7x.

PyPOTS lazy-loads single samples from HDF5 via `self.file_handle["X"][idx]`, and training DataLoaders use `shuffle=True`. That means shuffled training can repeatedly pull very large HDF5 chunks just to read individual samples, causing avoidable I/O/cache thrash.

## Relevant code

- `src/imputation_training/data_export.py`
  - `export_splits_to_h5(...)` creates `X` and `X_ori` with `chunks=(min(chunk_size, n_samples), n_timesteps, 19)`
- `configs/training/h5_export/default.yaml`
  - default `chunk_size: 1000`
- PyPOTS local behavior observed in installed package
  - HDF5 sample reads use `file_handle["X"][idx]`
  - imputation model training DataLoaders use `shuffle=True`

## Why this matters

The HDF5 layout is write-efficient but poorly matched to shuffled per-sample training reads. At paper scale this can slow training startup/epochs and increase filesystem pressure, especially on shared cluster storage.

## Possible approach

- Change the default HDF5 row chunk size to better match training read batches, e.g. 32-128 rows rather than 1000.
- Consider separate export defaults for sequential validation/test versus shuffled training.
- Optionally expose a target chunk byte size instead of row count so `n_days` is accounted for automatically.
- Benchmark epoch throughput before/after on representative train/val H5 files.

## Acceptance criteria

- Default chunks are small enough that single-sample or shuffled-batch reads do not pull ~100 MB+ chunks.
- Training throughput is measured before/after on a representative PyPOTS run.
- Existing H5 export and PyPOTS fit tests continue to pass.
